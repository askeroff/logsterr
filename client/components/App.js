import React from 'react';
import Layout from './Layout';

const App = () => (
  <Layout>
    <h1 className="page-title">TimeTracker</h1>
  </Layout>
);

export default App;
