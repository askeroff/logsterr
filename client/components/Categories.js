import React from 'react';
import Layout from './layout/Layout';

const Categories = () => (
  <Layout>
    <h1 className="page-title">Categories</h1>
    <p>This page will list all the categories now available;</p>
    <p>as well as let the user add new ones </p>
  </Layout>
);

export default Categories;
