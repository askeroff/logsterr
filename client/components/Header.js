import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logout from './Logout';
import '../styles/header.styl';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.guestLinks = this.guestLinks.bind(this);
  }

  guestLinks() {
    if (!this.props.user) {
      return (
        <ul className="header-menu">
          <li className="header-menu-item">
            <Link to="/">Home</Link>
          </li>
          <li className="header-menu-item">
            <Link to="/signup">Sign Up</Link>
          </li>
          <li className="header-menu-item">
            <Link to="/login">Log In</Link>
          </li>
        </ul>
      );
    }
    return (
      <ul className="header-menu">
        <li className="header-menu-item">
          <Link to="/">Home</Link>
        </li>
        <Logout history={this.props.history} />
      </ul>
    );
  }

  render() {
    return (
      <header>
        { this.guestLinks() }
      </header>
    );
  }
}

Header.defaultProps = {
  user: null,
  history: {},
};

Header.propTypes = {
  user: PropTypes.object,
  history: PropTypes.object,
};

export default Header;
