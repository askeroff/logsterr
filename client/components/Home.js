import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.user.loggedIn === true) {
      this.setState({ loggedIn: true });
    } else if (nextProps.user.loggedIn === false) {
      this.props.history.push('/login');
    }
  }

  shouldRender() {
    if (this.state.loggedIn) {
      return <Dashboard />;
    }
    return <Spinner />;
  }

  render() {
    return (
      <Layout>
        {this.shouldRender()}
      </Layout>
    );
  }
}

Home.propTypes = {
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(Home);
