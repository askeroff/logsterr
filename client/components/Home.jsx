"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var React = require("react");
var react_redux_1 = require("react-redux");
var Layout_1 = require("./layout/Layout");
var Index_1 = require("./dashboard/Index");
var Spinner_1 = require("./layout/Spinner");
var LandingPage_1 = require("./LandingPage");
var Home = /** @class */ (function (_super) {
    __extends(Home, _super);
    function Home() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.shouldRender = function () {
            var hasHistory = Object.prototype.hasOwnProperty.call(_this.props.history, 'push');
            if (_this.props.user.loggedIn) {
                return (<Layout_1.default showToGuests>
          <Index_1.default />
        </Layout_1.default>);
            }
            else if (_this.props.user.loggedIn === false && hasHistory) {
                return (<Layout_1.default showToGuests contentClass="landing-wrapper">
          <LandingPage_1.default />
        </Layout_1.default>);
            }
            return (<Layout_1.default showToGuests>
        <Spinner_1.default />
      </Layout_1.default>);
        };
        return _this;
    }
    Home.prototype.render = function () {
        return this.shouldRender();
    };
    Home.defaultProps = { history: {}, user: {} };
    return Home;
}(React.Component));
var mapStateToProps = function (state) { return ({
    user: state.user
}); };
exports["default"] = react_redux_1.connect(mapStateToProps)(Home);
