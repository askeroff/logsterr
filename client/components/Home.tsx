import * as React from 'react';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';
import { IUser } from '../types';
import LandingPage from './LandingPage';

interface Props {
  user: IUser,
  history: any
}

class Home extends React.Component<Props, any> {
  static defaultProps = { history: {}, user: {} };

  shouldRender = () => {
    const hasHistory = Object.prototype.hasOwnProperty.call(
      this.props.history,
      'push'
    );
    if (this.props.user.loggedIn) {
      return (
        <Layout showToGuests>
          <Dashboard />
        </Layout>
      );
    } else if (this.props.user.loggedIn === false && hasHistory) {
      return (
        <Layout showToGuests contentClass="landing-wrapper">
          <LandingPage />
        </Layout>
      );
    }
    return (
      <Layout showToGuests>
        <Spinner />
      </Layout>
    );
  };

  render() {
    return this.shouldRender();
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Home);
