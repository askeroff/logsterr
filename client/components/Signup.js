import React from 'react';
import Layout from './Layout';

const Signup = () => (
  <Layout>
    <h1>Signup Page</h1>
  </Layout>
);

export default Signup;
