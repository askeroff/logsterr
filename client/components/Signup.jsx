/* eslint react/no-array-index-key: off */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signUp } from '../actions/user';
import Layout from './layout/Layout';
import SignupForm from './layout/SignupForm';
import Spinner from './layout/Spinner';

class Signup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      invite: '',
      errors: [],
      spinner: false
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.user.email !== undefined) {
      this.props.history.push('/');
    }
    if (nextProps.user.errors && nextProps.user.errors.length > 0) {
      this.setState({ spinner: false, errors: nextProps.user.errors });
    }
  }

  handleEmailChange = event => {
    this.setState({ email: event.target.value });
  };

  handlePasswordChange = event => {
    this.setState({ password: event.target.value });
  };

  handleInviteValue = event => {
    this.setState({ invite: event.target.value });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.setState({
      spinner: true,
      errors: []
    });
    this.props.handleSignup({
      email: this.state.email,
      password: this.state.password,
      invite: this.state.invite
    });
  };

  renderErrors = () =>
    this.state.errors.map((message, index) => (
      <div key={index} className="server-response error">
        {message}
      </div>
    ));

  render() {
    return (
      <Layout>
        <h1 className="page-title">Signup Page</h1>
        <SignupForm
          myClassName="form__signup"
          handleSubmit={this.handleSubmit}
          emailValue={this.state.email}
          inviteValue={this.state.invite}
          handleInviteChange={this.handleInviteValue}
          passwordValue={this.state.password}
          handleEmailChange={this.handleEmailChange}
          handlePasswordChange={this.handlePasswordChange}
        />
        <div>{this.renderErrors()}</div>
        {this.state.spinner ? <Spinner /> : null}
      </Layout>
    );
  }
}

Signup.propTypes = {
  handleSignup: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  handleSignup(user) {
    dispatch(signUp(user));
  }
});

export const UnwrappedSignup = Signup;
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup);
