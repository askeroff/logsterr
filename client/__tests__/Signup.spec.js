import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { UnwrappedSignup } from '../components/Signup';

Enzyme.configure({ adapter: new Adapter() });

test.skip('Signup Component renders correctly', () => {
  const component = shallow(
    <UnwrappedSignup
      handleSignup={() => 1}
      user={{ email: 'test@gmail.com' }}
      history={{}}
    />
  );
  expect(component).toMatchSnapshot();
});
