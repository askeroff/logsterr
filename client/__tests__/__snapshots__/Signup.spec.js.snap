// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component renders correctly 1`] = `
<Connect(Layout)>
  <h1
    className="page-title"
  >
    Signup Page
  </h1>
  <LoginForm
    emailValue=""
    handleEmailChange={[Function]}
    handlePasswordChange={[Function]}
    handleSubmit={[Function]}
    myClassName="form signup"
    passwordValue=""
  />
  <div />
</Connect(Layout)>
`;
